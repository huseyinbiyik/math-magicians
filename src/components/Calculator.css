.calculator-container {
  display: flex;
  align-items: flex-start;
  width: 90%;
  margin: 100px auto;
  justify-content: space-between;
}

.calculator-welcome-text {
  display: inline;
  width: 30%;
  margin: 0;
  padding: 0;
}

.wrapper {
  margin-right: 300px;
  display: grid;
  grid-template-columns: auto auto auto auto;
  box-shadow:
    rgba(0, 0, 0, 0.25) 0 54px 55px,
    rgba(0, 0, 0, 0.12) 0 -12px 30px,
    rgba(0, 0, 0, 0.12) 0 4px 6px,
    rgba(0, 0, 0, 0.17) 0 12px 13px,
    rgba(0, 0, 0, 0.09) 0 -3px 5px;
}

.wrapper > button {
  border: 0.5px solid #8586936a;
  text-align: center;
  padding: 30px 35px;
}

.total {
  padding: 25px 35px;
  grid-column-start: 1;
  grid-column-end: 5;
  display: flex;
  justify-content: flex-end;
  background-color: #858693;
  color: white;
}

.zero {
  grid-column-start: 1;
  grid-column-end: 3;
}

.ac,
.plus-minus,
.percent,
.seven,
.eight,
.nine,
.four,
.five,
.six,
.one,
.two,
.three,
.zero,
.point {
  background-color: #e0e0e0;
}

.divide,
.multiply,
.minus,
.plus,
.equal {
  background-color: #f4913e;
}
