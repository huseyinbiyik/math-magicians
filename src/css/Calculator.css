.wrapper {
  margin-top: 100px;
  display: grid;
  grid-template-columns: auto auto auto auto;
  box-shadow: rgba(0, 0, 0, 0.25) 0 54px 55px, rgba(0, 0, 0, 0.12) 0 -12px 30px,
    rgba(0, 0, 0, 0.12) 0 4px 6px, rgba(0, 0, 0, 0.17) 0 12px 13px,
    rgba(0, 0, 0, 0.09) 0 -3px 5px;
}

.wrapper > button {
  border: 0.5px solid #8586936a;
  text-align: center;
  padding: 30px 35px;
}

.total {
  padding: 25px 35px;
  grid-column-start: 1;
  grid-column-end: 5;
  display: flex;
  justify-content: flex-end;
  background-color: #858693;
  color: white;
}

.zero {
  grid-column-start: 1;
  grid-column-end: 3;
}

.ac,
.plus-minus,
.percent,
.seven,
.eight,
.nine,
.four,
.five,
.six,
.one,
.two,
.three,
.zero,
.point {
  background-color: #e0e0e0;
}

.divide,
.multiply,
.minus,
.plus,
.equal {
  background-color: #f4913e;
}
