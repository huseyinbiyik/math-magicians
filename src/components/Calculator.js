import React, { useState } from "react";
import "../css/Calculator.css";
import calculate from "../logic/calculate";

const Calculator = () => {
  const [state, setState] = useState({ total: 0 });

  const onClickHandler = (e) => {
    setState(() => calculate(state, e.target.innerText));
  };

  const { total, operation, next } = state;
  return (
    <section className="calculator-container">
      <h2 className="calculator-welcome-text">Let's do some math! ⇢ ⇢ ⇢</h2>
      <div className="wrapper">
        <div className="total">
          {total}
          {operation}
          {next}
        </div>
        <button className="ac" type="button" onClick={onClickHandler}>
          AC
        </button>
        <button className="plus-minus" type="button" onClick={onClickHandler}>
          +/-
        </button>
        <button className="percent" type="button" onClick={onClickHandler}>
          %
        </button>
        <button className="divide" type="button" onClick={onClickHandler}>
          ÷
        </button>
        <button className="seven" type="button" onClick={onClickHandler}>
          7
        </button>
        <button className="eight" type="button" onClick={onClickHandler}>
          8
        </button>
        <button className="nine" type="button" onClick={onClickHandler}>
          9
        </button>
        <button className="multiply" type="button" onClick={onClickHandler}>
          x
        </button>
        <button className="four" type="button" onClick={onClickHandler}>
          4
        </button>
        <button className="five" type="button" onClick={onClickHandler}>
          5
        </button>
        <button className="six" type="button" onClick={onClickHandler}>
          6
        </button>
        <button className="minus" type="button" onClick={onClickHandler}>
          -
        </button>
        <button className="one" type="button" onClick={onClickHandler}>
          1
        </button>
        <button className="two" type="button" onClick={onClickHandler}>
          2
        </button>
        <button className="three" type="button" onClick={onClickHandler}>
          3
        </button>
        <button className="plus" type="button" onClick={onClickHandler}>
          +
        </button>
        <button className="zero" type="button" onClick={onClickHandler}>
          0
        </button>
        <button className="point" type="button" onClick={onClickHandler}>
          .
        </button>
        <button className="equal" type="button" onClick={onClickHandler}>
          =
        </button>
      </div>
    </section>
  );
};

export default Calculator;
