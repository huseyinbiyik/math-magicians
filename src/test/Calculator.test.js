import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

describe('Test for Calculator page', () => {
  it('It should render Calculator component', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
