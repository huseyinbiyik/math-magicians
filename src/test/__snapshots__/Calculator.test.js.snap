// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Calculator page It should render Calculator component 1`] = `
<section
  className="calculator-container"
>
  <h2
    className="calculator-welcome-text"
  >
    Lets do some math!
  </h2>
  <div
    className="wrapper"
  >
    <div
      className="total"
    >
      0
    </div>
    <button
      className="ac"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="plus-minus"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="percent"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="divide"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="seven"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="eight"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="nine"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="multiply"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="four"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="five"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="six"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="minus"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="one"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="two"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="three"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="plus"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="zero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="point"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="equal"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</section>
`;
